tests/test_sim2.py

```diff
@@ -89,7 +89,37 @@ def test_compose():
     assert wSw == imgSw.compose(wSimg)
 
 
-def test_inverse():
+def test_compose_2() -> None:
+    """Verify composition of Sim2 objects works for non-identity input."""
+    aSb = Sim2(R=rotmat2d(np.deg2rad(90)), t=np.array([1, 2]), s=4)
+
+    bSc = Sim2(R=rotmat2d(np.deg2rad(-45)), t=np.array([3, 4]), s=0.5)
+
+    aSc = aSb.compose(bSc)
+    # Via composition: 90 + -45 = 45 degrees
+    assert aSc.theta_deg == 45.0
+    # Via composition: 4 * 0.5 = 2.0
+    assert aSc.scale == 2.0
+
+
+def test_compose_3() -> None:
+    """Verify correctness of composed inverted and non-inverted transformations."""
+
+    # Note: these are dummy translation values; should be ignored when considering correctness
+    aSb = Sim2(R=rotmat2d(np.deg2rad(20)), t=np.array([1, 2]), s=2)
+
+    bSc = Sim2(R=rotmat2d(np.deg2rad(30)), t=np.array([1, 2]), s=3)
+
+    aSc = Sim2(R=rotmat2d(np.deg2rad(50)), t=np.array([1, 2]), s=6)
+
+    # ground truth is an identity transformation
+    aSa_gt = Sim2(R=np.eye(2), t=np.zeros(2), s=1.0)
+    aSa_est = aSb.compose(bSc).compose(aSc.inverse())
+
+    assert np.isclose(aSa_gt.theta_deg, aSa_est.theta_deg, atol=1e-5)
+
+
+def test_inverse() -> None:
     """ """
```

Comment:
Could we fill this with something brief?

Reply:
Thanks for the review Ben. Sure, just added that.